internal/pkg/aws/s3/s3.go

```diff
@@ -214,21 +216,22 @@ func (s *S3) upload(bucket, key string, buf io.Reader) (string, error) {
 // [s3:// URI]: https://docs.aws.amazon.com/AmazonS3/latest/userguide/access-bucket-intro.html#accessing-a-bucket-using-S3-format
 func parseS3URI(uri string) (bucket, key string, err error) {
 	parsed := strings.SplitN(strings.TrimPrefix(uri, s3URIPrefix), "/", 2)
-	if len(parsed) != 2 {
+	bucket = parsed[0]
+	if bucket == "" {
 		return "", "", fmt.Errorf("cannot parse S3 URI %s into bucket name and key", uri)
 	}
-	return parsed[0], parsed[1], nil
+	if len(parsed) == 2 {
+		key = parsed[1]
+	}
+	return
 }
 
-// parseObjectURL parses the bucket name and object key from a [virtual-hosted-style access URL].
+// parseHTTPURL parses the bucket name and optional key from a [virtual-hosted-style access URL].
 // For example: https://DOC-EXAMPLE-BUCKET1.s3.us-west-2.amazonaws.com/puppy.png
 //
 // [virtual-hosted-style access URL]: https://docs.aws.amazon.com/AmazonS3/latest/userguide/access-bucket-intro.html#virtual-host-style-url-ex
-func parseObjectURL(url string) (bucket, key string, err error) {
+func parseHTTPURL(url string) (bucket, key string, err error) {
```

Comment:
```suggestion
func parseHTTPURI(url string) (bucket, key string, err error) {
```
nit: to match the `parseS3URI` naming?